client_code/routing/__init__.py

```diff
@@ -7,25 +7,194 @@
 
 __version__ = "1.9.0"
 
-from ._routing import (
-    add_to_cache,
-    clear_cache,
-    error_form,
-    get_cache,
+from anvil.js import window as _w
+
+from . import _navigation
+from . import _router as _r
+from ._decorators import error_form, route, template
+from ._logging import log as _log
+from ._logging import logger
```

Comment:
I think the _logging module needs moving into utils (and incorporating into the _timed module too)

Reply:
good idea

Reply:
I'll make a separate PR for that